### PR TITLE
Improve flexibility

### PR DESCRIPTION
**Pull request by GuillaumeDSM**
Goal:
1. Be able to use nexus while under http (local test purposes)
2. Upload files which are stored under a different name than the ones to be uploaded

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -259,33 +260,39 @@ api.upload_maven_component(
   version: VERSION,
   repository: REPOSITORY_NAME,
   tag: TAG,                     # Optional
+  content: FILE_CONTENT,        # Optional
```
Comment (FrancisLevesque) [inline README.md:263]: Can you rename `content` to `remote_filename`, `upstream_filename`, or something along those lines? I think `content` is pretty misleading.

Comment (FrancisLevesque) [reply README.md:263]: And add a description of the field to the comment above this method description.

**Comment by GuillaumeDSM** [comment]
> @GuillaumeDSM Thanks for the changes :) Once my comments are addressed and you add some tests, I'll add these changes into the next minor version.

It should be fine now :)

**Review by FrancisLevesque — APPROVED** [review]
I've got some minor thoughts around wording in the README and some additional tests, but I'll add those in another PR and create the patch version of the gem after.

**Pull request merged**